ha/core/config/consul_kv_store.py

```diff
@@ -0,0 +1,185 @@
+#!/usr/bin/env python3
+
+# Copyright (c) 2021 Seagate Technology LLC and/or its Affiliates
+#
+# This program is free software: you can redistribute it and/or modify it under the
+# terms of the GNU Affero General Public License as published by the Free Software
+# Foundation, either version 3 of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+# PARTICULAR PURPOSE. See the GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License along
+# with this program. If not, see <https://www.gnu.org/licenses/>. For any questions
+# about this software or licensing, please email opensource@seagate.com or
+# cortx-questions@seagate.com.
+
+import consul
+import errno
+import socket
+from urllib.parse import urlparse
+
+class ConsulKvStore:
+    """ Represents a Consul kv Store """
+
+    name = "consul"
+    version = "/v1"
+
+    def __init__(self, url: str):
+        """
+        Consul KV store.
+
+        Args:
+            url (str): Consul url with prefix
+
+        Example:
+            ConsulKvStore("consul://localhost:8500/cortx/ha/system/config")
```

Comment:
Version should be after /cortx/ha like /cortx/ha/v1/system/config.
/cortx/ha is fixed prefix for HA module. Version again should be fixed for a release. These could be stored in config file and prefixed automatically.

Reply:
Changed format

only that not sure if we put version in conf then it may bind to ha and implemented in way that it will not use ha anywhere.

Should we keep in ha.conf

Reply:
updated version and putted in const